client/App.zig

```diff
@@ -1105,6 +1111,14 @@ pub fn frameLoop(self: *App) !void {
 
             // send a frame start if the engine thread is waiting for us to do so
             if (self.game.load_state.full_init and begin_new_frame) {
+                const dropped_files = try self.game.input.takeDroppedFiles(self.gpa);
+                defer if (dropped_files) |files| {
+                    for (files.paths) |file| {
+                        self.gpa.free(file);
+                    }
+                    self.gpa.free(files.paths);
+                };
```

Comment:
I hate this. Is there a better way or do string lists always have to be managed like this?

Reply:
You could wrap the type or create a custom deinit function if you *really* want (like the process args stuff does), but it's not really necessary, this is fine.

Reply:
Generally we can avoid allocations in the first place but I'm guessing we can't really do that here since there's 2 layers of unknown lengths

Reply:
We can't easily do so here, no. Since this is an uncommon thing to be hitting, I don't mind the small code wart to pointing out "hey! there's uncommon allocation patterns going on here, take care when refactoring". If we start to need to do this more frequently, then we should abstract it though.